src/main/java/com/google/javascript/cl2dts/DeclarationGenerator.java

```diff
@@ -211,12 +253,28 @@ protected void printSummary() {}
 
   public String produceDts(Compiler compiler) {
     Set<String> provides = new HashSet<>();
-    // TODO: only inspect inputs which were explicitly provided by the user
+    Set<String> transitiveProvides = new HashSet<>();
+    List<String> depgraphRoots = parseDepgraphRoots();
```

Comment:
But why make it a warning? If the user didn't specify a depgraph, that's just what we do, right?

OTOH, if the user did specify a depgraph and we weren't able to parse anything out of it, we should given an error and fail.